packages/renderer/src/stores/providers.ts

```diff
@@ -78,3 +79,29 @@ export async function fetchProviders(): Promise<ProviderInfo[]> {
   });
   return result;
 }
+
+/**
+ * Counting the container connection per type is a small but
+ * expensive iterative operation, creating a derived readable to avoid
+ * recomputing it each time we need it.
+ */
+export const containerConnectionCount = derived([providerInfos], ([$providerInfos]) => {
```

Comment:
This code LGTM

An alternative suggestion(non-blocking, personal preference) could also be: 


```ts

export const containerConnectionCount = derived([providerInfos], ([$providerInfos]) => {
  const connections = $providerInfos.flatMap(({ containerConnections }) =>
    containerConnections.filter(({ status }) => status !== 'stopped'),
  );
  return {
    podman: connections.filter(({ type }) => type === 'podman').length,
    docker: connections.filter(({ type }) => type === 'docker').length,
  };
});

```